tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,662 @@
+            prod_attrs = entities.Product(id=prod_id).read_json()
+            self.assertIn(repo_id, [
+                repo['id'] for repo in prod_attrs['repositories']])
+
+    @run_only_on('sat')
+    def test_create_multiple_docker_repo_multiple_products(self):
+        """@Test: Create multiple Docker-type repositories on multiple products.
+
+        @Assert: Multiple docker repositories are created with a Docker image
+        and they all belong to their respective products.
+
+        @Feature: Docker
+
+        """
+        for _ in range(0, randint(1, 5)):
+            prod_id = entities.Product(organization=self.org_id).create()['id']
+            for _ in range(0, randint(1, 3)):
+                repository = _create_repository(prod_id, None, None)
+                repo_id = repository['id']
+                prod_attrs = entities.Product(id=prod_id).read_json()
+                self.assertIn(repo_id, [
+                    repo['id'] for repo in prod_attrs['repositories']])
+
+    @run_only_on('sat')
+    def test_sync_docker_repo(self):
+        """@Test: Create and sync a Docker-type repository
+
+        @Assert: A repository is created with a Docker repository
+        and it is synchronized.
+
+        @Feature: Docker
+
+        """
+        prod_id = entities.Product(organization=self.org_id).create()['id']
+        repository = _create_repository(prod_id, None, None)
+        repo_id = repository['id']
+
+        task_result = entities.Repository(id=repo_id).sync()['result']
+        self.assertEqual(
+            u'success',
+            task_result)
+        attrs = entities.Repository(id=repo_id).read_json()
+        self.assertGreaterEqual(attrs[u'content_counts'][u'docker_image'], 1)
+
+    @run_only_on('sat')
+    @data(
+        gen_string('alpha', 15),
+        gen_string('alphanumeric', 15),
+        gen_string('numeric', 15),
+        gen_string('latin1', 15),
+        gen_string('utf8', 15),
+        gen_string('html', 15),
+    )
+    def test_update_docker_repo_name(self, new_name):
+        """@Test: Create a Docker-type repository and update its name.
+
+        @Assert: A repository is created with a Docker image and that its
+        name can be updated.
+
+        @Feature: Docker
+
+        """
+        name = u'Busybox'
+        upstream_name = u'busybox'
+        content_type = u'docker'
+        prod_id = entities.Product(organization=self.org_id).create()['id']
+
+        repository = _create_repository(prod_id, name, upstream_name)
+        real_attrs = entities.Repository(id=repository['id']).read_json()
+        self.assertEqual(real_attrs['name'], name)
+        self.assertEqual(real_attrs['docker_upstream_name'], upstream_name)
+        self.assertEqual(real_attrs['content_type'], content_type)
+
+        # Update the repository name to random value
+        real_attrs['name'] = new_name
+        client.put(
+            entities.Repository(id=repository['id']).path(),
+            real_attrs,
```

Comment:
When updating an entity, it should be possible to provide values only for the attributes that are being updated. For example:

```python
client.put(
    entities.Repository(id=repository['id']).path(),
    {'name': new_name},
    verify=False,
    auth=get_server_credentials(),
).raise_for_status()
```